src/top/data/bbox_reg_util.py

```diff
@@ -0,0 +1,152 @@
+"""
+Reference:
+    https://github.com/skhadem/3D-BoundingBox/blob/master/torch_lib/ClassAverages.py
+    https://github.com/skhadem/3D-BoundingBox/blob/master/torch_lib/Dataset.py
+"""
+
+
+from dataclasses import dataclass
+from typing import Tuple
+import numpy as np
+import os
+import json
+from simple_parsing.helpers.serialization.serializable import Serializable
+
+import torch as th
+import cv2
+from scipy.spatial.transform import Rotation as R
+
+from top.data.schema import Schema
+from top.run.box_generator import Box
+
+"""
+Enables writing json with numpy arrays to file
+"""
+class NumpyEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, np.ndarray):
+            return obj.tolist()
+        return json.JSONEncoder.default(self,obj)
+
+
+"""
+Class will hold the average dimension for a class, regressed value is the residual
+"""
+class ClassAverages:
+    def __init__(self, classes=[]):
+        self.dimension_map = {}
+        self.filename = os.path.abspath(os.path.dirname(__file__)) + '/class_averages.txt'
```

Comment:
prefer `json` prefix for json-encoded files
```suggestion
        self.filename = os.path.abspath(os.path.dirname(__file__)) + '/class_averages.json'
```